### ISSUE TITLE
Binder with DateField is not persisting null values after validation error

### ISSUE THREAD

**Issue by mlopezFC** [vaadin/framework#11723]
### Vaadin Framework version
Tested with 8.9.0
### Browser version
Chrome Version 76.0.3809.132 
### Web container name and version
Jetty 9.3.9.v20160517
### Description of the bug
By using a `DateField` with `Binder`, with a non-required field, cleaning the value will persist a null to the Bean, but if the value is cleared after a validation/conversion error, then the null is not persisted to the Bean.
### Minimal reproducible example
[See attached project](https://github.com/vaadin/framework/files/3617695/datefield-problem.zip)
Video showing the problem:
![DateTimeFieldProblem2](https://user-images.githubusercontent.com/28312758/64984279-dd3e7f80-d898-11e9-9036-d8e2c1a1611b.gif)
The last part of the video shows that after clearing the DateField and loosing focus, the null value is not persisted.
### Expected behavior
The null value should be written to the underlying bean after a validation error.
### Actual behavior
The null value is not persisted to the underlying bean after a validation error. If a valid value is filled in and then cleared then the null value is persisted.

**Comment by TatuLund** [comment]
This needs to be retested after https://github.com/vaadin/framework/pull/11758

**Comment by TatuLund** [comment]
Could be related to https://github.com/vaadin/framework/issues/9754

**Comment by mlopezFC** [comment]
I re-tested it with Vaadin 8.9.3 and the same thing still happens, and yes, it seems to be similar to what is happening in #9754.

**Comment by mlopezFC** [comment]
I tried the same thing with Vaadin 14, and DatePicker, and the behavior is different: when entering an invalid value and losing focus, a null is persisted in the bean, so obviously this problem does not happen, but on the other hand I'm not sure if it's ok to persist a null, given that validation was unsuccessful.

**Comment by TatuLund** [comment]
#11758 is not included I re-tested it with Vaadin 8.9.3 , but is in 8.10-SNAPSHOT.

**Comment by TatuLund** [comment]
> I tried the same thing with Vaadin 14, and DatePicker, and the behavior is different

Yes, the change is a cherry pick from Vaadin 14 Binder, and I would assume 8.10-SNAPTSHOT to behave in same manner.

**Comment by mlopezFC** [comment]
Ok, was able to try this with 8.10-SNAPSHOT and the same thing described in this issue still happens:

1. Select a date using the DateField and then lose focus
2. The Date is going to be persisted on the bean so the label will show the Date
3. Delete the date and then lose focus
4. The null will be persisted on the bean so the label will show "null"
5. Select a date using the DateField and then lose focus again
6. Replace the date with an invalid value and then lose focus. It will show a validation error
7. Delete the invalid value and then lose focus again: The null will NOT be persisted after the validation error was shown
